src/olympia/shelves/serializers.py

```diff
@@ -45,7 +45,10 @@ def get_addons(self, obj):
             request = self.context.get('request', None)
             request.GET = request.GET.copy()
             request.GET.update(params)
-            return AddonSearchView(request=request).data
+            addons = AddonSearchView(request=request).data
+            for key in request.GET.dict():
+                del request.GET[key]
```

Comment:
`request.GET = QueryDict()` would do to get an empty GET querydict instead of looping through the keys.

But it'd miss the case where we want to pass on some parameters from the `/shelves/` endpoint (e.g. `/shelves/?lang=fr` to enable French localisation) down to each of the requests.  So we've got to keep a copy of the unmodified query params from the original request object before we start changing it, so we're starting with those parameters each time.